ATP.BusinessLogicLayer/GenericRepo/FlightRepository.cs

```diff
@@ -0,0 +1,62 @@
+﻿using ATP.DataAccessLayer.IGenericRepo;
+using ATP.DataAccessLayer.Models;
+using CsvHelper;
+using System.Globalization;
+
+namespace ATP.BusinessLogicLayer.GenericRepo;
+public class FlightRepository : IGenericRepo<Flight>
+{ 
+    private List<Flight> _flights;
+
+    public FlightRepository(string csvFilePath)
+    {
+        _flights = LoadFlightsFromCsv(csvFilePath);
+    }
+    private List<Flight> LoadFlightsFromCsv(string csvFilePath)
+    {
+        List<Flight> flights = new List<Flight>();
+
+        using (var reader = new StreamReader(csvFilePath))
+        using (var csv = new CsvReader(reader, CultureInfo.InvariantCulture))
+        {
+            flights = csv.GetRecords<Flight>().ToList();
+        }
+
+        return flights;
+    }
+    public Flight GetById(int id)
+    {
+        return _flights.FirstOrDefault(f => f.Id == id);
+    }
+
+    public void Add(Flight entity)
+    {
+        entity.Id = _flights.Count > 0 ? _flights.Max(f => f.Id) + 1 : 1;
+        _flights.Add(entity);
+    }
+
```

Comment:
This doesn't add to the csv file. Add, update, delete should effect the database, which is csv on our case.